src/main/java/crazypants/enderio/conduit/me/MEConduit.java

```diff
@@ -0,0 +1,339 @@
+  public static TextureRegistry.TextureSupplier longTextureD = TextureRegistry.registerTexture("blocks/meConduitDense");
+
+  private boolean isDense;
+  private int playerID = -1;
+
+  public MEConduit() {
+    this(0);
+  }
+  
+  public MEConduit(int itemDamage) {
+    isDense = itemDamage == 1;
+  }
+
+  @SideOnly(Side.CLIENT)
+  public static void initIcons() {
+  }
+
+  public static int getDamageForState(boolean isDense) {
+    return isDense ? 1 : 0;
+  }
+
+  @Override
+  public Class<? extends IConduit> getBaseConduitType() {
+    return IMEConduit.class;
+  }
+
+  @Override
+  public ItemStack createItem() {
+    return new ItemStack(itemMEConduit.getItem(), 1, getDamageForState(isDense));
+  }
+
+  @Override
+  public AbstractConduitNetwork<?, ?> getNetwork() {
+    return network;
+  }
+
+  @Override
+  public boolean setNetwork(AbstractConduitNetwork<?, ?> network) {
+    this.network = (MEConduitNetwork) network;
+    return true;
+  }
+
+  @Override
+  public void writeToNBT(NBTTagCompound nbtRoot) {
+    super.writeToNBT(nbtRoot);
+    nbtRoot.setBoolean("isDense", isDense);
+    nbtRoot.setInteger("playerID", playerID);
+  }
+
+  @Override
+  public void readFromNBT(NBTTagCompound nbtRoot, short nbtVersion) {
+    super.readFromNBT(nbtRoot, nbtVersion);
+    isDense = nbtRoot.getBoolean("isDense");
+    if(nbtRoot.hasKey("playerID")) {
+      playerID = nbtRoot.getInteger("playerID");
+    } else {
+      playerID = -1;
+    }
+  }
+
+  public void setPlayerID(int playerID) {
+    this.playerID = playerID;
+  }
+
+  @Override
+  public int getChannelsInUse() {
+    int channelsInUse = 0;
+    IGridNode node = getNode();
+    if(node != null) {
+      for (IGridConnection gc : node.getConnections()) {
+        channelsInUse = Math.max(channelsInUse, gc.getUsedChannels());
+      }
+    }
+    return channelsInUse;
+  }
+
+  @Override
+  @Method(modid = "appliedenergistics2")
+  public boolean canConnectToExternal(EnumFacing dir, boolean ignoreDisabled) {
+    World world = getBundle().getBundleWorldObj();
+    BlockCoord pos = getLocation();
+    TileEntity te = world.getTileEntity(pos.getLocation(dir).getBlockPos());
+
+    if(te instanceof TileConduitBundle) {
+      return false;
+    }
+
+    // because the AE2 API doesn't allow an easy query like "which side can connect to an ME cable" it needs this mess
+    if(te instanceof IPartHost) {
+      IPart part = ((IPartHost) te).getPart(dir.getOpposite());
+      if(part == null) {
+        part = ((IPartHost) te).getPart(AEPartLocation.INTERNAL);
+        return part != null;
+      }
+      if(part.getExternalFacingNode() != null) {
+        return true;
+      }
+      String name = part.getClass().getSimpleName();
+      return "PartP2PTunnelME".equals(name) || "PartQuartzFiber".endsWith(name) ||
+              "PartToggleBus".equals(name) || "PartInvertedToggleBus".equals(name);
+    } else if(te instanceof IGridHost) {
```

Comment:
if you're using eclipse, please import the code formater and import order setting (top folder of the project) and run them on these files. If not, tell me and I'll do it after merging the PR.